### PR TITLE
Clear cache when historyCacheSize is set to 0

### LINKED ISSUE #1221: History cache not cleared after historyCacheSize has been set to 0 and cache was not empty
If the history cache is not empty and the `historyCacheSize` is set to 0, the history cache is not cleared, but still used. This leads to very subtle bugs.

https://github.com/bigskysoftware/htmx/blob/7ea35d03c6226aaf278b47d80170132bad9384f0/src/htmx.js#L1945-L1956

### PR DESCRIPTION
**Pull request by aomader**
Fixes #1221

### CODE REVIEW

**Comment by Telroshan** [comment]
Hey, if you're still interested in merging those changes, could you please retarget your PR to the `dev` branch ?

File: src/htmx.js [hunk 1]
```diff
@@ -2193,6 +2193,7 @@ return (function () {
                     historyCache.shift(); // shrink the cache and retry
                 }
             }
+            if (historyCache.length === 0) localStorage.removeItem("htmx-history-cache");
```
Comment (1cg) [inline src/htmx.js:2196]: can you add a comment here?

Also maybe this should be up near the top and avoid all the logic in this method?

**Review by 1cg — CHANGES_REQUESTED** [review]

File: src/htmx.js (hunk 1 above)
Comment (aomader) [reply src/htmx.js]: @1cg I added a comment and moved it to the top to exit early. Note however that this changes the current behavior w.r.t. the `htmx:historyItemCreated` that was triggered up until now. However, I'd argue that the new behavior is more correct.

Comment (1cg) [reply src/htmx.js]: agree

**Pull request merged**